docs/src/main/asciidoc/spring-cloud-commons.adoc

```diff
@@ -167,6 +167,8 @@ This feature addresses the problem of stateful beans that only get their configu
 For instance, if a `DataSource` has open connections when the database URL is changed via the `Environment`, you probably want the holders of those connections to be able to complete what they are doing.
 Then, the next time something borrows a connection from the pool, it gets one with the new URL.
 
+Sometimes, it might even be mandatory to apply `@RefreshScope` annotation on some beans which can be only initialized once. For example, the default Spring Boot DataSource is now `com.zaxxer.hikari.HikariDataSource`, which by default is `refreshable` (defined in `RefreshScopeBeanDefinitionEnhancer`), as HikariDataSource can no longer be modified once it's started. In order to refresh this kind of `immutable` beans, you will have to either annotate the beans with `@RefreshScope` or specify the list of class names under the property key `spring.cloud.refresh.extra-refreshable`.
```

Comment:
I would write it this way (to break up that one long sentence in the middle and a few smaller changes):

Sometimes, it might even be mandatory to apply `@RefreshScope` annotation on some beans which can be only initialized once. For example, the default Spring Boot DataSource is now `com.zaxxer.hikari.HikariDataSource`. By default, `HikariDataSource` is `refreshable` (defined in `RefreshScopeBeanDefinitionEnhancer`), as `HikariDataSource` can no longer be modified once it has been started. To refresh this kind of `immutable` bean, you have to either annotate the beans with `@RefreshScope` or specify the list of class names under the property key (`spring.cloud.refresh.extra-refreshable`).